process/block/shardblock.go

```diff
@@ -351,24 +358,53 @@ func (sp *shardProcessor) RestoreBlockIntoPools(headerHandler data.HeaderHandler
 	return nil
 }
 
-func (sp *shardProcessor) restoreMetaBlockIntoPool(miniBlockHashes map[int][]byte) error {
+func (sp *shardProcessor) restoreMetaBlockIntoPool(miniBlockHashes map[int][]byte, metaBlockHashes [][]byte) error {
 	metaBlockPool := sp.dataPool.MetaBlocks()
 	if metaBlockPool == nil {
 		return process.ErrNilMetaBlockPool
 	}
 
+	for _, metaBlockHash := range metaBlockHashes {
+		buff, err := sp.store.Get(dataRetriever.MetaBlockUnit, metaBlockHash)
+		if err != nil {
+			continue
+		}
+
+		metaBlock := block.MetaBlock{}
+		err = sp.marshalizer.Unmarshal(&metaBlock, buff)
+		if err != nil {
+			log.Error(err.Error())
+			continue
+		}
+
+		metaBlockPool.Put(metaBlockHash, &metaBlock)
+
+		err = sp.store.GetStorer(dataRetriever.MetaBlockUnit).Remove(metaBlockHash)
+		if err != nil {
+			log.Error(err.Error())
+		}
+
+		nonceToByteSlice := sp.uint64Converter.ToByteSlice(metaBlock.Nonce)
```

Comment:
put into the metahdr nonce pool as well.

Reply:
Done